File: ros2_ws/src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

node_name = 'smartphone'

class SmartphoneNode(Node):
    def __init__(self):
        super().__init__(f'{node_name}')
        self.subscriber_ = self.create_subscription(String, 'robot_news', self.callback_robot_news, 10)
        self.get_logger().info(f'{node_name} has started...')

    def callback_robot_news(self, msg):
        self.get_logger().info(f'New msg: {msg.data}')

def main(args=None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: ros2_ws/src/my_cpp_pkg/src/smartphone.cpp
#include <string_view>

#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"

namespace {
std::string_view sv_name{"smartphone"};
}

class SmartphoneNode : public rclcpp::Node {
 public:
  SmartphoneNode() : Node(sv_name.data()) {
    RCLCPP_INFO(this->get_logger(), "%s has started...", sv_name.data());
    subscriber_ = this->create_subscription<example_interfaces::msg::String>(
        "robot_news", 10,
        std::bind(&SmartphoneNode::callbackRobotNews, this,
                  std::placeholders::_1));
  }

 private:
  void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg) {
    RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
  }
  rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<SmartphoneNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}